.github/workflows/build.yml

```
@@ -1,24 +1,43 @@
-name: build
-on: push
+name: Build
+on:
+  push:
+    branches:
+      - main
 
 jobs:
   build:
-    runs-on: ubuntu-20.04
-    container:
-      image: openjdk:8-jdk
+    strategy:
+      matrix:
+        java: [
+          1.8, # Minimum
```

```suggestion
          8, # Minimum
```
Required for `actions/setup-java@v2`